subiquity/ui/views/snaplist.py

```diff
@@ -463,10 +463,13 @@ def make_main_screen(self, data):
                 "package, publisher and versions available."))
 
     def done(self, sender=None):
-        log.debug("snaps to install %s", self.selections_by_name)
-        self.controller.done(sorted(
-            self.selections_by_name.values(),
-            key=lambda s: s.name))
+        if getattr(self, "selections_by_name", None):
+            log.debug("snaps to install %s", self.selections_by_name)
+            self.controller.done(sorted(
+                self.selections_by_name.values(),
+                key=lambda s: s.name))
+        else:
+            self.controller.done([])
```

Comment:
Um. I don't understand the flow that gets to this path, but probably just setting `self.selections_by_name` to {} in `__init__` is easier?

Reply:
Yes, that's much more sensible